content/blog/trustless-lightning-payments-for-content.md

```diff
@@ -0,0 +1,357 @@
+* Bob decodes the payment request string to get the payment point of the invoice, call it `p3`.
+* Bob calculates `s2*G`. If it is equal to `p3 - p1`, then Bob knows that the invoice is valid.
+* Bob pays the Lightning invoice, and gets the value of `s1 + s2` as the preimage.
+* Bob calculates `s1 = (s1 + s2) - s2` to get the decryption key, and decrypts the content.
+
+If Carol tries to give Bob an invalid invoice, Bob will know before he pays the invoice.
+
+The above example assumes that Bob is able to trust that the value of `p1` is valid.
+
+{{< center-figure src="/post-data/trustless-lightning-payments/ptlc-example-2.png" caption="" height=600 width=600 >}}
+
+Now any node in the network can relay content from any creator, and earn a profit by
+selling it trustlessly to any consumer. The consumer does not have to trust that the
+relay is an honest actor, and vice versa. The only requirement is that the payment
+point generated by the original content creator must be publicly known and trusted.
+
+
+#### Payment for content with Squeak Protocol
+
+The [Squeak Protocol](https://github.com/squeaknode/squeak/blob/master/docs/PROTOCOL.md)
+is a social media protocol that utilizes the properties of PTLCs on
+lightning to create a trustless, decentralized alternative to Twitter.
+
+With the Squeak Protocol, any user can download an encrypted post (called a "squeak"),
+and verify several properties of the squeak:
+
+1) The squeak has a unique hash that is the same across all nodes
+2) The squeak was created and signed by a certain person.
+3) The squeak was created after a certain time.
+4) The squeak was created in reply to another squeak (or not).
+5) The squeak was encrypted using a scalar that corresponds to a certain
+elliptic curve point.
+
+After these properties have been verified, the consumer can request an invoice from
+the seller that will unlock the decrypted content of the squeak (after payment):
+
+{{< center-figure src="/post-data/trustless-lightning-payments/locked-squeak.png" caption="" height=600 width=600 >}}
+
+If the consumer then clicks the "buy" button, they will be presented with an offer
+that they received from the seller:
+
+{{< center-figure src="/post-data/trustless-lightning-payments/received-offer.png" caption="" height=600 width=600 >}}
+
+Then, if the consumer decides that the squeak is worth the price, they can pay the
+invoice to unlock the squeak:
+
+{{< center-figure src="/post-data/trustless-lightning-payments/unlocked-squeak.png" caption="" height=600 width=600 >}}
+
+The consumer now has a copy of the unlocked squeak on their own node, and
+they can now re-sell the squeak to other nodes in the network.
+
+#### How squeaks work under the hood
+
+There are two components in a squeak:
+
+* The squeak header
+* The external fields outside the header.
+
+A squeak header has the following fields:
+
+Field Size | Description | Data type | Comments
```

Comment:
I'll see if I can get the tables to render a bit nicer :)

Reply:
I just fixed the table formatting.